glui.cpp

```diff
@@ -210,7 +264,17 @@ void GLUI_Main::setup_default_glut_callbacks( void )
   glutPassiveMotionFunc( glui_passive_motion_func );
   glutEntryFunc( glui_entry_func );
   glutVisibilityFunc( glui_visibility_func );
-  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */
+  /*  glutIdleFunc( glui_idle_func );    // FIXME!  100% CPU usage!      */  
+  //This is needed to host the X11 copy-paste selection-owner.
+  //It's really no big deal. The scheduler will divide time up
+  //and unless "vsync" is off a continuously rendering client
+  //will wait for the "vertical-blank."
+  //GLUT isn't a good fit for passive applications, and it is
+  //historically for full-motion 3-D displays.
+  //Clients can figure out a wait solution in their Idle function.
+  //freeglut provides a default idle, which it does not expose,
+  //but it's open-souce and so copy-friendly.
+  glutIdleFunc( glui_idle_func );
```

Comment:
It depends on what projects are using GLUI and for what. It certainly doesn't depend on what projects are using GLUT.

If software is passive it shouldn't rely on GLUT for that, because there's nothing in GLUT that guarantees how threads will yield inside the idle. Software needs to do that for itself. 

The best that can be done is to provide a default Idle for if none are provided that implements something like freeglut's wait for events. But I'm not even sure that's the best option. And in X11 mode freeglut I believe is waiting for XEvents, which is not really an option because there are two event cues, and so doing that would starve the Copy and Paste event queue, causing software that is pasting to wait for the GLUI application to wake up before the paste will finish!

My advise would be stay with the previous version of GLUI or fix it in your application. In reality there is unlikely to be a new version of GLUI officially released for a very long time, and if code is using the unreleased version then that code has to be up to speed.

About the only simple fix would be to go back to not supplying an idle procedure if GLUI_GLX is not defined. But frankly that was a lot of nonsense to maintain. Code should be approached first and foremost from a maintenance perspective.